auth/auth-server/src/server/gas_sponsorship/contract_interaction/base.rs

```diff
@@ -64,32 +75,82 @@ fn sponsored_malleable_atomic_match_calldata(
     refund_native_eth: bool,
     refund_amount: U256,
     signature: Bytes,
-) -> Result<sponsorMalleableAtomicMatchSettleCall, AuthServerError> {
-    let processMalleableAtomicMatchSettleCall {
-        quoteAmount,
-        baseAmount,
-        receiver,
-        internalPartyPayload,
-        matchSettleStatement,
-        proofs,
-        linkingProofs,
-    } = processMalleableAtomicMatchSettleCall::abi_decode(calldata)
+    use_malleable_match_connector: bool,
+) -> Result<Bytes, AuthServerError> {
+    // Decode the original call
+    let original_call = processMalleableAtomicMatchSettleCall::abi_decode(calldata)
         .map_err(AuthServerError::gas_sponsorship)?;
 
-    Ok(sponsorMalleableAtomicMatchSettleCall {
-        quoteAmount,
-        baseAmount,
-        receiver,
-        internalPartyMatchPayload: internalPartyPayload,
-        malleableMatchSettleStatement: matchSettleStatement,
-        matchProofs: proofs,
-        matchLinkingProofs: linkingProofs,
+    // If the user requested their order to be routed through the connector,
+    // use calldata for the connector's ABI
+    if use_malleable_match_connector {
+        return sponsored_malleable_atomic_match_calldata_with_connector(
+            original_call,
+            refund_address,
+            nonce,
+            refund_native_eth,
+            refund_amount,
+            signature,
+        );
+    }
+
+    // Otherwise, route directly to the gas sponsor
+    let call = sponsorMalleableAtomicMatchSettleCall {
+        quoteAmount: original_call.quoteAmount,
+        baseAmount: original_call.baseAmount,
+        receiver: original_call.receiver,
+        internalPartyMatchPayload: original_call.internalPartyPayload,
+        malleableMatchSettleStatement: original_call.matchSettleStatement,
+        matchProofs: original_call.proofs,
+        matchLinkingProofs: original_call.linkingProofs,
         refundAddress: refund_address,
         nonce,
         refundNativeEth: refund_native_eth,
         refundAmount: refund_amount,
         signature,
-    })
+    };
+    Ok(call.abi_encode().into())
+}
+
+/// Convert a `processMalleableAtomicMatchSettle` calldata into an
+/// `executeMalleableAtomicMatchWithInputCall` call
```

Comment:
Do you think it's worthwhile to add a `WithOutput` call to set `outputAmount` as well, or better to wait until a customer specifically requests it?